src/Uno.UI/UI/Xaml/Controls/ScrollContentPresenter/NativeScrollContentPresenter.macOS.cs

```diff
@@ -76,7 +76,14 @@ public void SetContentOffset(CGPoint contentOffset, bool animated)
 
 		partial void OnContentChanged(NSView previousView, NSView newView)
 		{
+			if(previousView != null) 
+			{
+				previousView.SetParent(null);
+			}
+
 			DocumentView = newView;
+
+			newView.SetParent(TemplatedParent);
```

Comment:
This makes sense, I wonder why we seemingly haven't needed to do it in `NativeScrollContentPresenter` for Android or iOS, though.

Reply:
This is not needed on iOS/Android because the native ScrollViewer relies on the Children property, not on a `DocumentView` property. Let's add this as a comment.

```suggestion
			// This is not needed on iOS/Android because the native ScrollViewer
			// relies on the Children property, not on a `DocumentView` property.
			newView.SetParent(TemplatedParent);
```